2020/day-2-password-philosophy/index.ts

```diff
@@ -0,0 +1,33 @@
+const parseList = (list: string) => {
+  return list
+    .trim()
+    .split('\n')
+    .map(entry => entry.split(': '))
+    .map(([policy, password]) => {
+      const [range, char] = policy.trim().split(' ')
+      const [min, max] = range.split('-').map(Number)
+
+      return { min, max, char, password }
+    })
+}
+
+export const getValidPasswords1 = (list: string) => {
+  const entries = parseList(list)
+
+  return entries.filter(({ min, max, char, password }) => {
+    const charCount = Array.from(password.matchAll(new RegExp(char, 'g'))).length
```

Comment:
Oh this is really neat!

Reply:
I think `.matchAll` returned an iterator or something, so `Array.from` converts it into a standard array. Could also use the spread: `[...password.matchAll(new RegExp(char, 'g'))].length`